cvat/apps/engine/ddln/tasks/vls_lines/models.py

```diff
@@ -0,0 +1,139 @@
+        self.lat_vanishing_point = None
+
+    @property
+    def track_id(self):
+        return self.id
+
+    def calculate_vanishing_points(self, reporter):
+        lon_lines = [self.left_line, self.center_line, self.right_line]
+        self.lon_vanishing_point = self._calculate_vanishing_point(lon_lines, reporter, is_lon=True)
+        lat_lines = [self.start_line, self.designator_line, self.end_line]
+        if all(line is not None for line in lat_lines):
+            self.lat_vanishing_point = self._calculate_vanishing_point(lat_lines, reporter, is_lon=False)
+
+    def fix_order(self, reporter):
+        if self.lon_vanishing_point:
+            self._check_lon_order(reporter)
+        else:
+            self._check_parallel_lon_order(reporter)
+        self._check_lat_order(reporter)
+
+    def apply_visibility(
+        self,
+        left_visible,
+        right_visible,
+        center_visible,
+        start_visible,
+        end_visible,
+        designator_visible
+    ):
+        if not left_visible:
+            self.left_line = None
+        if not right_visible:
+            self.right_line = None
+        if not center_visible:
+            self.center_line = None
+        if not start_visible:
+            self.start_line = None
+        if not end_visible:
+            self.end_line = None
+        if not designator_visible:
+            self.designator_line = None
+
+    def _check_lon_order(self, reporter):
+        original = [self.left_line, self.center_line, self.right_line]
+        max_angle = 0
+        edges = None
+        for a, b in combinations(original, 2):
+            angle = get_lines_angle(a, b)
+            if angle > max_angle:
+                max_angle = angle
+                edges = [a, b]
+        new_center = next(filter(lambda line: line not in edges, original))
+        new_left, new_right = edges
+        if get_counterclockwise_angle(new_right.get_angle(), new_left.get_angle()) >= math.pi:
+            new_left, new_right = new_right, new_left
+        final = [new_left, new_center, new_right]
+        if final != original:
+            reporter.report_lon_disorder()
+
+    def _check_parallel_lon_order(self, reporter):
+        large_step = abs(self.left_line.c - self.right_line.c)
+        small_step_a = abs(self.center_line.c - self.left_line.c)
+        small_step_b = abs(self.center_line.c - self.right_line.c)
+        if small_step_a >= large_step or small_step_b >= large_step:
+            reporter.report_lon_disorder()
+
+    def _check_lat_order(self, reporter):
+        original = [self.end_line, self.designator_line, self.start_line]
+        if any(line is None for line in original):
+            return
+        distant_point = self.lon_vanishing_point
+        if not distant_point:
+            # if longitudinal lines are parallel, just assume they are directed away from the camera
+            distant_point = Line.by_point_and_angle(Point(0, 0), 0).intersect(self.center_line)
```

Comment:
I don't understand this computation. What if centerline parallel to image axis?

Reply:
Then `distant_point` is set to `Point(0, -10000)` (some distant point from the camera) in the next `if`. 

When longitudinal lines are parallel, it's impossible to identify the direction of the runway, but often the runway is directed away from the camera, so this assumption allow this check to work in frequent cases (but leads to false-positives in some cases). These checks are just warnings to annotators which they can address or ignore.
Depending on the feedback of the annotators this check can be improved or removed altogether.